packages/jupyter-ai/jupyter_ai/chat_handlers/base.py

```diff
@@ -9,15 +9,39 @@
 from jupyter_ai.config_manager import ConfigManager, Logger
 from jupyter_ai.models import AgentChatMessage, HumanChatMessage
 from jupyter_ai_magics.providers import BaseProvider
+from traitlets.config import Configurable
 
 if TYPE_CHECKING:
     from jupyter_ai.handlers import RootChatHandler
 
 
-class BaseChatHandler:
+class BaseChatHandler(Configurable):
```

Comment:
Not sure if configurable, which makes this a Traitlet and entry points work well together. For example, configurable properties are usually managed at runtime via runtime attributes at Jupyter startup  (jupyter lab --attribute) or via a config file. It's not clear how making these available as entry points will help with configuring these.